transformer_engine/debug/features/fake_quant_fp8.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) 2022-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""FakeQuantFp8 Feature support for nvidia-dlframework-inspect"""
+
+from typing import Optional
+
+import torch
+
+import nvdlfw_inspect.api as debug_api
+from nvdlfw_inspect.registry import Registry, api_method
+from nvdlfw_inspect.utils import append_parent_docstring
+
+
+import transformer_engine_torch as tex
+from transformer_engine.debug.features.api import TEConfigAPIMapper
+from transformer_engine.common.recipe import Format
+from transformer_engine.pytorch.tensor import Quantizer
+from transformer_engine.pytorch.tensor.float8_tensor import Float8Quantizer
+from transformer_engine.pytorch.tensor.mxfp8_tensor import MXFP8Quantizer
+from transformer_engine.pytorch.fp8 import _default_sf_compute
+
+
+def fake_quantize_fp8(tensor: torch.Tensor, fp8_format: tex.DType, margin=0, out=None):
```

Comment:
Could we make the name more general and just do fake_quantize instead? We will have other datatypes soon like FP4 and we will want to be able to use the same mechanism there.